cf/data/data.py

```diff
@@ -5665,7 +5651,7 @@ def all(self, axis=None, keepdims=True, split_every=None):
 
         """
         d = self.copy(array=False)
-        dx = self._get_dask()
+        dx = self.to_dask_array()
```

Comment:
I am seeing weird-looking errors from this, and the same error for the equivalent `any` (but notably not `__bool__`) change:

```python
======================================================================
ERROR: test_Data_any (__main__.DataTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_Data.py", line 1490, in test_Data_any
    self.assertFalse(d.any())
  File "/home/sadie/anaconda3/envs/cf-env/lib/python3.8/unittest/case.py", line 757, in assertFalse
    if expr:
  File "/home/sadie/cf-python/cf/data/data.py", line 848, in __bool__
    return bool(self.to_dask_array())
  File "/home/sadie/anaconda3/envs/cf-env/lib/python3.8/site-packages/dask/array/core.py", line 1792, in __bool__
    return bool(self.compute())
  File "/home/sadie/anaconda3/envs/cf-env/lib/python3.8/site-packages/dask/base.py", line 292, in compute
    (result,) = compute(self, traverse=False, **kwargs)
  File "/home/sadie/anaconda3/envs/cf-env/lib/python3.8/site-packages/dask/base.py", line 575, in compute
    results = schedule(dsk, keys, **kwargs)
  File "/home/sadie/anaconda3/envs/cf-env/lib/python3.8/site-packages/dask/local.py", line 554, in get_sync
    return get_async(
  File "/home/sadie/anaconda3/envs/cf-env/lib/python3.8/site-packages/dask/local.py", line 497, in get_async
    for key, res_info, failed in queue_get(queue).result():
  File "/home/sadie/anaconda3/envs/cf-env/lib/python3.8/concurrent/futures/_base.py", line 437, in result
    return self.__get_result()
  File "/home/sadie/anaconda3/envs/cf-env/lib/python3.8/concurrent/futures/_base.py", line 389, in __get_result
    raise self._exception
  File "/home/sadie/anaconda3/envs/cf-env/lib/python3.8/site-packages/dask/local.py", line 539, in submit
    fut.set_result(fn(*args, **kwargs))
...
...
  File "/home/sadie/anaconda3/envs/cf-env/lib/python3.8/site-packages/dask/core.py", line 119, in _execute_task
    return func(*(_execute_task(a, cache) for a in args))
  File "/home/sadie/anaconda3/envs/cf-env/lib/python3.8/site-packages/dask/array/slicing.py", line 2157, in setitem
    x[tuple(indices)] = v
  File "/home/sadie/anaconda3/envs/cf-env/lib/python3.8/site-packages/numpy/ma/core.py", line 3399, in __setitem__
    np.copyto(dindx, dval, where=~mindx)
  File "<__array_function__ internals>", line 180, in copyto
TypeError: Cannot cast scalar from dtype('float64') to dtype('int64') according to the rule 'same_kind'
```

Are those issues relating to the order we are merging these PRs, i.e. something fixed by #388, I imagine? Just checking in case this is an issue to be addressed, not related to the migration of `varray` that concerns this PR in particular...

Reply:
Aha. This is looks like what https://github.com/dask/dask/issues/9026 fixes.

Reply:
Ah, fair. Thanks for clarifying. It sounds like I still have some environment tweaking to do, then. All good, marking as resolved.